### PR TITLE
feat: Product list

### PR DESCRIPTION
**Pull request by AtulHande03a**
- made product list page
- made api call to the products and displayed
- made product context 
- recreated category db and made api calls
- created category context

### CODE REVIEW

File: src/components/Hero.jsx [hunk 1]
```diff
@@ -15,9 +16,9 @@ export const Hero = () => {
       <div className="banner-detail p-0-5 flex-column flex-center">
         <h1 className="banner-heading pb-0-5">Modular Furniture</h1>
         <p className="banner-offer pb-0-5">Extra 30% off</p>
-        <a href="./pages/product-list.html" class="btn btn-primary">
+        <Link to="products" class="btn btn-primary">
```
Comment (horsemaker) [inline src/components/Hero.jsx:19]: I think giving `/` here will be safer.

File: src/assets/images/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import beds from "../images/bed.jpg";
+import chair from "../images/chair.jpg";
+import coffeeTable from "../images/coffe-table.jpg";
+import sofa from "../images/sofa.jpg";
+
+export { beds, chair, coffeeTable, sofa };
```
Comment (horsemaker) [inline src/assets/images/index.js:6]: If possible shift the images to the `public` folder.

File: src/components/Categories.jsx [hunk 3]
```diff
@@ -1,28 +1,33 @@
-import { categories } from "data.js";
+// import { categories } from "data.js";
```
Comment (t007rushi) [inline src/components/Categories.jsx:1]: remove comments

File: src/components/ProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    discount,
+    shipsIn,
+  } = product;
+
+  return (
+    <>
+      <article className="card-verticle flex flex-column mr-2 my-1 mb-2">
+        <section className="image-container mb-0-5">
+          <img
+            loading="lazy"
```
Comment (horsemaker) [inline src/components/ProductCard.jsx:21]: Lazy loading! Great! 👌

File: src/contexts/product-context.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const [productList, setProductList] = useState([]);
+
+  const getProducts = async () => {
+    try {
+      setLoading(true);
+      const res = await axios.get("./api/products");
+      setProductList((_prev) => res.data.products);
+      setLoading(false);
+    } catch (error) {
+      console.log(error);
```
Comment (horsemaker) [inline src/contexts/product-context.js:17]: I guess, loading should be set to false here too

**Review by horsemaker — APPROVED** [review]

File: src/contexts/product-context.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+  const [loading, setLoading] = useState(false);
+  const [productList, setProductList] = useState([]);
+
+  const getProducts = async () => {
+    try {
+      setLoading(true);
+      const res = await axios.get("./api/products");
```
Comment (t007rushi) [inline src/contexts/product-context.js:13]: destructure here itself {data}

**Review by t007rushi — APPROVED** [review]
Liked that you did useDocumnetTitle Hook 
Great job

File: src/assets/images/index.js (hunk 2 above)
Comment (AtulHande03a) [reply src/assets/images/index.js:6]: will shift once P0 submission is done

File: src/components/Categories.jsx (hunk 3 above)
Comment (AtulHande03a) [reply src/components/Categories.jsx:1]: done

File: src/contexts/product-context.js (hunk 5 above)
Comment (AtulHande03a) [reply src/contexts/product-context.js:17]: done

File: src/components/Hero.jsx (hunk 1 above)
Comment (AtulHande03a) [reply src/components/Hero.jsx:19]: done

**Pull request merged**